plugins/com.google.cloud.tools.eclipse.util/src/com/google/cloud/tools/eclipse/util/MavenUtils.java

```diff
@@ -63,12 +81,62 @@ public static boolean hasMavenNature(IProject project) {
     }
   }
 
-  public static Artifact resolveArtifact(IProgressMonitor monitor, String groupId,
-      String artifactId, String type, String version, String classifier,
-      List<ArtifactRepository> repositories) throws CoreException {
-    Artifact artifact = MavenPlugin.getMaven().resolve(groupId, artifactId, version, type,
-        classifier, repositories, monitor);
-    return artifact;
+  public static Artifact resolveArtifact(
+      String groupId,
+      String artifactId,
+      String type,
+      String version,
+      String classifier,
+      List<ArtifactRepository> repositories,
+      IProgressMonitor monitor)
+      throws CoreException {
+    return runOperation(
+        monitor,
+        (context, system, progress) -> {
+          Artifact artifact =
+              MavenPlugin.getMaven()
+                  .resolve(groupId, artifactId, version, type, classifier, repositories, progress);
+          return artifact;
+        });
+  }
+
+  /**
+   * Perform some Maven-related action that may result in a change to the local Maven repositories,
+   * ensuring that required {@link ISchedulingRule scheduling rules} are held.
+   *
+   * @param ruleMonitor a progress monitor to be used when waiting to obtain the rule
+   * @param rule the rule that must be held
```

Comment:
`ruleMonitor` --> `monitor`, and there is no `rule` parameter. Also, `operation` sounds better than `supplier`?